examples/all-sky/rrtmgp_allsky.F90

```diff
@@ -1,390 +1,422 @@
-      call stop_on_err(k_dist%gas_optics(p_lay, p_lev, &
-                                         t_lay, t_sfc, &
-                                         gas_concs,    &
-                                         atmos,        &
-                                         lw_sources,   &
-                                         tlev = t_lev))
-      call stop_on_err(clouds%increment(atmos))
-      call stop_on_err(rte_lw(atmos, top_at_1, &
-                              lw_sources,      &
-                              emis_sfc,        &
-                              fluxes))
-      !$acc exit data delete(lw_sources%lay_source, lw_sources%lev_source_inc, lw_sources%lev_source_dec, lw_sources%sfc_source, lw_sources)
-    else
-      !$acc enter data create(toa_flux)
-      fluxes%flux_dn_dir => flux_dir(:,:)
-
-      call stop_on_err(k_dist%gas_optics(p_lay, p_lev, &
-                                         t_lay,        &
-                                         gas_concs,    &
-                                         atmos,        &
-                                         toa_flux))
-      call stop_on_err(clouds%delta_scale())
-      call stop_on_err(clouds%increment(atmos))
-      call stop_on_err(rte_sw(atmos, top_at_1, &
-                              mu0,   toa_flux, &
-                              sfc_alb_dir, sfc_alb_dif, &
-                              fluxes))
-      !$acc exit data delete(toa_flux)
-    end if
-    !print *, "******************************************************************"
-    call system_clock(finish, clock_rate)
-    elapsed(iloop) = finish - start
-  end do
-  !
-  call system_clock(finish_all, clock_rate)
-  !
-  !$acc exit data delete(lwp, iwp, rel, rei)
-  !$acc exit data delete(p_lay, p_lev, t_lay, t_lev)
-
-#ifdef _OPENACC
-  avg = sum( elapsed(merge(2,1,nloops>1):) ) / real(merge(nloops-1,nloops,nloops>1))
-
-  print *, "Execution times - min(s)        :", minval(elapsed) / real(clock_rate)
-  print *, "                - avg(s)        :", avg / real(clock_rate)
-  print *, "                - per column(ms):", avg / real(ncol) / (1.0e-3*clock_rate)
-#else
-  print *, "Execution times - total(s)      :", (finish_all-start_all) / real(clock_rate)
-  print *, "                - per column(ms):", (finish_all-start_all) / real(ncol*nloops) / (1.0e-3*clock_rate)
-#endif
-
-  if(is_lw) then
-    !$acc exit data copyout(flux_up, flux_dn)
-    if(write_fluxes) call write_lw_fluxes(input_file, flux_up, flux_dn)
-    !$acc exit data delete(t_sfc, emis_sfc)
-  else
-    !$acc exit data copyout(flux_up, flux_dn, flux_dir)
-    if(write_fluxes) call write_sw_fluxes(input_file, flux_up, flux_dn, flux_dir)
-    !$acc exit data delete(sfc_alb_dir, sfc_alb_dif, mu0)
-  end if
-  !$acc enter data create(lwp, iwp, rel, rei)
-end program rte_rrtmgp_clouds
+subroutine stop_on_err(error_msg)
```

Comment:
In every other file the diffs show only the addition of the OpenMP directives. This files shows up as completely changed, which is confusing. Do you know why?

Reply:
Correct, its because of the carriage returns. I have a specific commit for this already in this PR:
https://github.com/earth-system-radiation/rte-rrtmgp/pull/99/commits/c03b95f4acc24028070c0ab7b5ab16ffd651110e

Note that when those `^M` are removed it will always look like the entire file changed.

Unfortunately, this file has become a headche because of the `^M`. (I spent almost half a day dealing with the best way to handle it already).

I think the cleanest way to do this is as follows. 
1. close this PR
2. create a separate PR for the control `^M`
3. Merge the PR in for the control `^M`
3. rebase/merge my changes into the tip of the new develop branch
4. create a new PR

Another possibility, would be:
1. close this PR
2. Rebase my feature branch to remove the commit with the control `^M`.
2. This will cause issues with other commits being applied, but hopefully this won't be too bad.
3. Push the changes into my feature branch

Let me know what you would like to do.

Reply:
You have figured out how to remove the `^M` characters now? If you explain how I could update my branch, so that the PR shows only the OpenMP additions.

Reply:
@naromero77 This question is for you

Reply:
On Linux it is super easy with this utility. 
`https://linux.die.net/man/1/dos2unix`

If you are on Mac and use homebrew, do:
`brew install dos2unix`

It should be as easy as: `dos2unix rrtmgp_allsky.F90 rrtmgp_allsky.F90` should do it.

Reply:
Our last messages crossed paths -- let me know what you would like me to do next here.

If you update your develop branch with the `^M` change, it *might* create a conflict on this PR. It is also possible the git will be smart enough to figure out how to do the merge cleanly an only show the OpenMP changes in the PR which is what you want. I have run into this issue in the past before and unfortunately git-merge had a tough time with it.

Reply:
Alternatively, if its just a matter of reviewing the OpenMP changes, then most of the changes for this `rrtmgp_allsky.f90` are in this commit:
https://github.com/earth-system-radiation/rte-rrtmgp/pull/99/commits/0e937fe5081da526ae2e81f6dbc9fd4bd8764c6f

And a few more here:
https://github.com/earth-system-radiation/rte-rrtmgp/pull/99/commits/35e6997b3c32ebe269d0a2e0a9bb522ac5ac6f9f

Reply:
Well, despite having `dos2unix`ed the file in a separate commit on my end the merge is still showing the entire file as changed. Shrug.

Reply:
Another idea that I have.  Do not change anything on your end, but I will try deleting the `^M` specific commit in my feature branch and then do a rebase. I can try that on a local test branch and see if it works first. Worse case scenario is that it could confuse the compiler during the parsing process and lead to a CI failure.

Reply:
But anytime you remove `^M`, it will show the entire file as changed. There is no way around that AFAIK.